src/SMU-Agilent_B29xx/main.py

```diff
@@ -390,16 +399,43 @@ def apply(self):
         value = str("{:.4E}".format(self.value))  # makes sure that self.value fits into SCPI command in terms of length
         if self.pulse:
             # get channel ready at specified values
+ 
             self.port.write(":SOUR%s:%s %s" % (self.channel, self.commands[self.source], self.pulseofflevel))
-            # arming the pulse trigger
-            self.port.write(":SOUR%s:%s:TRIG %s" % (self.channel, self.commands[self.source], value))
+    
+            # arming the pulse trigger; mis-using the sweep function with identical start and stop values so we can use number-of-pulses functionality
+            if self.source.startswith("Current"):
+                self.port.write(":SOUR%s:CURR:STAR %s" % (self.channel, value))
+                self.port.write(":SOUR%s:CURR:STOP %s" % (self.channel, value))
+            
+            if self.source.startswith("Voltage"):
+                self.port.write(":SOUR%s:VOLT:STAR %s" % (self.channel, value))
+                self.port.write(":SOUR%s:VOLT:STOP %s" % (self.channel, value))
+            
             # releasing the pulse trigger
             self.port.write(":INIT (@%s)" % self.channel)
+            
         else:
             # set output to specified values
             self.port.write(":SOUR%s:%s %s" % (self.channel, self.commands[self.source], value))
```

Comment:
I wonder that this works because the Sweep mode is now also set for non-pulse measurements. If Sweep mode would be only used for pulses, it would be more clear that this commands works as it is typically used with mode FIX.